h/traversal/group_membership.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+
+from pyramid.httpexceptions import HTTPNotFound
+
+from h.exceptions import InvalidUserId
+from h.models import Group, GroupMembership, User
+
+
+@dataclass
+class GroupMembershipContext:
+    group: Group
+    user: User
+    membership: GroupMembership | None
```

Comment:
This is the new `context` object that will be passed to security permissions predicate functions and to views for the `GET`, `POST`, `PATCH` and `DELETE` `/api/groups/{pubid}/members/{userid}` APIs. This is a more appropriate context object for group membership views: as well as providing convenient access to `context.group` and `context.user` it also provides the `membership` itself including `context.membership.roles`.

These views previously used [`GroupContext`](https://github.com/hypothesis/h/blob/7f53416154863cc080a3fb520bbe8b3324409b53/h/traversal/group.py#L7-L11) and this new `GroupMembershipContext` is backwards-compatible with `GroupContext` (it has the same `group` attribute and adds `user` and `membership`) so changing to this context class shouldn't break anything.